chia/cmds/wallet_funcs.py

```diff
@@ -284,22 +284,65 @@ async def make_offer(args: dict, wallet_client: WalletRpcClient, fingerprint: in
         print("Not creating offer: Must be offering and requesting at least one asset")
     else:
         offer_dict: Dict[Union[uint32, str], int] = {}
+        driver_dict: Dict[str, Any] = {}
         printable_dict: Dict[str, Tuple[str, int, int]] = {}  # Dict[asset_name, Tuple[amount, unit, multiplier]]
+        nft_warning: bool = False
         for item in [*offers, *requests]:
-            wallet_id, amount = tuple(item.split(":")[0:2])
-            if int(wallet_id) == 1:
-                name: str = "XCH"
-                unit: int = units["chia"]
-            else:
-                name = await wallet_client.get_cat_name(wallet_id)
-                unit = units["cat"]
+            name, amount = tuple(item.split(":")[0:2])
+            try:
+                id: Union[uint32, str] = bytes32.from_hexstr(name).hex()
+                unit = 1
+            except ValueError:
+                try:
+                    hrp, _ = bech32_decode(name)
+                    if hrp == "nft":
+                        coin_id = decode_puzzle_hash(name)
+                        unit = 1
+                        info = NFTInfo.from_json_dict((await wallet_client.get_nft_info(coin_id.hex()))["nft_info"])
+                        nft_warning = True
+                        id = info.launcher_id.hex()
+                        assert isinstance(id, str)
+                        if item in requests:
+                            driver_dict[id] = {
+                                "type": "singleton",
+                                "launcher_id": "0x" + id,
+                                "launcher_ph": "0x" + info.launcher_puzhash.hex(),
+                                "also": {
+                                    "type": "metadata",
+                                    "metadata": info.chain_info,
+                                    "updater_hash": "0x" + info.updater_puzhash.hex(),
+                                },
+                            }
+                            if info.supports_did:
+                                driver_dict[id]["also"]["also"] = {
+                                    "type": "ownership",
+                                    "owner": "()",
+                                    "transfer_program": {
+                                        "type": "royalty transfer program",
+                                        "launcher_id": "0x" + info.launcher_id.hex(),
+                                        "royalty_address": "0x" + info.royalty_puzzle_hash.hex(),
+                                        "royalty_percentage": str(info.royalty_percentage),
+                                    },
+                                }
+                    else:
+                        id = decode_puzzle_hash(name).hex()
```

Comment:
What case is this handling? Some other bech32m-encoded thing?

Maybe a few comments would help clarify what each block of code is attempting.